frontend/src/metabase/components/Button.jsx

```diff
@@ -0,0 +1,35 @@
+import React, { Component, PropTypes } from "react";
+
+import Icon from "metabase/components/Icon.jsx";
+
+import cx from "classnames";
+
+import _ from "underscore";
+
+const BUTTON_VARIANTS = [
+    "small",
+    "medium",
+    "large",
+    "primary",
+    "warning",
+    "cancel",
+    "purple",
+    "borderless"
+];
+
+const Button = ({ className, icon, children, ...props }) => {
+    let variantClasses = BUTTON_VARIANTS.filter(variant => props[variant]).map(variant => "Button--" + variant);
```

Comment:
Nah wasn't suggesting that. Was just mentioning how there's the potential for that to happen and since it's being done via prop not an explicit CSS class it might be a bit harder to track down.